sdk/eventhub/event-hubs/src/connectionContext.ts

```diff
@@ -168,105 +174,61 @@ export namespace ConnectionContext {
-      if (!state.wasConnectionCloseCalled && (state.numSenders || state.numReceivers)) {
-        logger.verbose(
-          "[%s] connection.close() was not called from the sdk and there were some " +
-            "sender or receiver links or both. We should reconnect.",
-          connectionContext.connection.id
-        );
-        await delay(Constants.connectionReconnectDelay);
-        // reconnect senders if any
+      // Close all senders and receivers to ensure clean up of timers & other resources.
+      if (state.numSenders || state.numReceivers) {
         for (const senderName of Object.keys(connectionContext.senders)) {
           const sender = connectionContext.senders[senderName];
           if (!sender.isConnecting) {
-            logger.verbose(
-              "[%s] calling detached on sender '%s' with address '%s'.",
-              connectionContext.connection.id,
-              sender.name,
-              sender.address
-            );
-            sender.onDetached(connectionError || contextError).catch((err) => {
-              logger.verbose(
-                "[%s] An error occurred while reconnecting the sender '%s' with adress '%s' %O.",
-                connectionContext.connection.id,
-                sender.name,
-                sender.address,
-                err
-              );
-            });
-          } else {
-            logger.verbose(
-              "[%s] sender '%s' with address '%s' is already reconnecting. Hence not " +
-                "calling detached on the sender.",
-              connectionContext.connection.id,
-              sender.name,
-              sender.address
-            );
+            await sender.close();
           }
         }
-        // reconnect receivers if any
         for (const receiverName of Object.keys(connectionContext.receivers)) {
           const receiver = connectionContext.receivers[receiverName];
           if (!receiver.isConnecting) {
-            logger.verbose(
-              "[%s] calling detached on receiver '%s' with address '%s'.",
-              connectionContext.connection.id,
-              receiver.name,
-              receiver.address
-            );
-            receiver.onDetached(connectionError || contextError).catch((err) => {
-              logger.verbose(
-                "[%s] An error occurred while reconnecting the receiver '%s' with adress '%s' %O.",
-                connectionContext.connection.id,
-                receiver.name,
-                receiver.address,
-                err
-              );
-            });
-          } else {
-            logger.verbose(
-              "[%s] receiver '%s' with address '%s' is already reconnecting. Hence not " +
-                "calling detached on the receiver.",
-              connectionContext.connection.id,
-              receiver.name,
-              receiver.address
-            );
+            await receiver.close();
```

Comment:
Same comment about swallowing errors on close here (I'm not sure there's a way to surface these to the user, we could log if it fails to close as well).

Reply:
Fixed with https://github.com/Azure/azure-sdk-for-js/pull/6737/commits/06e9251b0941110ccdaa398a637d4fba818f69f1